pkg/ztunnel/reconciler/cell.go

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package reconciler
+
+import (
+	"time"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/statedb"
+	"github.com/cilium/statedb/reconciler"
+)
+
+var Cell = cell.Module(
+	"enrollment-reconciler",
+	"Reconciler for namespace enrollment for ztunnel mTLS",
+	cell.Provide(reconciler.NewExpVarMetrics),
+	cell.Provide(
+		NewEnrolledNamespacesTable,
+		NewEnrollmentReconciler,
+	),
+	cell.Invoke(registerEnrollmentReconciler),
+)
+
+func registerEnrollmentReconciler(
+	params reconciler.Params,
+	ops reconciler.Operations[*Namespace],
+	tbl statedb.RWTable[*Namespace],
+	m *reconciler.ExpVarMetrics,
+) error {
+	_, err := reconciler.Register(
+		params,
+		tbl,
+		(*Namespace).Clone,
+		(*Namespace).SetStatus,
+		(*Namespace).GetStatus,
+		ops,
+		nil, // no batch operations support
+
+		reconciler.WithMetrics(m),
+		reconciler.WithPruning(time.Minute),
```

Comment:
Given we don't implement a prune function this lets us avoid to call that function at all I believe
```suggestion
		reconciler.WithoutPruning(),
```